tools/android/packaging/xbmc/src/channels/SyncProgramsJobService.java.in

```diff
@@ -209,10 +209,14 @@ public class SyncProgramsJobService extends JobService
                         .insert(
                                 TvContractCompat.PreviewPrograms.CONTENT_URI,
                                 previewProgram.toContentValues());
-        long programId = ContentUris.parseId(programUri);
-        Log.d(TAG, "Inserted new program: " + programId);
-        media.setProgramId(programId);
-        mediasAdded.add(media);
+
+        if (programUri != null)
```

Comment:
Can you explain how this call can return null? According to the docs it must always return a long : https://developer.android.com/reference/android/content/ContentUris#parseId(android.net.Uri)

Likely this here is throwing an exception. Googling around led me to this: https://github.com/crosswalk-project/chromium-crosswalk/blob/master/chrome/android/java/src/org/chromium/chrome/browser/provider/ChromeBrowserProvider.java#L33 . Maybe what needs to be done is move this to an utils class, catch the proper expexceptions and return -1, then compare to -1 instead of null

Reply:
Its the previous command that possibly returns null

```
        Uri programUri =
                getContentResolver()
                        .insert(
                                TvContractCompat.PreviewPrograms.CONTENT_URI,
                                previewProgram.toContentValues());
```

https://developer.android.com/reference/android/content/ContentResolver#insert(android.net.Uri,%20android.content.ContentValues)

Reply:
Oh, 🤦  sorry for the noise. Was looking around for programId instead of programUri. All good of course